src/client/gui/CMakeLists.txt.linux

```diff
@@ -29,7 +29,7 @@ set(MULTIPASS_GUI_EXECUTABLE_FULL_PATH ${CMAKE_RUNTIME_OUTPUT_DIRECTORY}/${MULTI
 
 file(GLOB_RECURSE MULTIPASS_GUI_SOURCE_FILES ${MULTIPASS_GUI_SOURCE_DIR}/lib/*.dart)
 
-set(FLUTTER_BUILD_ARGS --release --suppress-analytics --verbose)
+set(FLUTTER_BUILD_ARGS --release --suppress-analytics --verbose --build-name=${MULTIPASS_VERSION})
```

Comment:
Shouldn't we split between build name and number?

Reply:
Ah, yes we should!

Reply:
Since the --build-number can only contain digits, I made it so that only the digits are extracted from the version string. So from `1.15.0-...`, all the digits are extracted. If there are none, it defaults to 0. This already happens by default on macOS, so I made it the same on all platforms.

Reply:
For my understanding, what would we get in these examples?

- 1.15.0-rc3.748+ga9411c0e9
- 1.15.0-rc4+win.3531+g65170ab8a.win
- 1.16.0-dev.3557.pr698+g9bae52f58.win
- 1.16.0-dev.3550.pr3811+g2c48add28.mac
- 1.16.0-dev.764.ci7330+gfa725c28c

Reply:
And didn't you say that we could have the entire thing in the build name? Also, notice that we don't actually have a build number anywhere in there. It's all deterministic regardless of the build.

Reply:
`1.15.0-rc3.748+ga9411c0e9` => `--build-name=1.15.0 --build-number=3748941109`
`1.15.0-rc4+win.3531+g65170ab8a.win` => `--build-name=1.15.0 --build-number=43531651708`
`1.16.0-dev.3557.pr698+g9bae52f58.win` => `--build-name=1.16.0 --build-number=355769895258`
`1.16.0-dev.3550.pr3811+g2c48add28.mac` => `--build-name=1.16.0 --build-number=3350381124828`
`1.16.0-dev.764.ci7330+gfa725c28c` => `--build-name=1.16.0 --build+-number=764733072528`

I was wrong about being able to pass something like `--build-name=1.16.0-dev.764.ci7330+gfa725c28c`. It worked on macos, firstly because for them the version is just a single string, not 3 numbers, and secondly because they do what I did up there aka removing all of the non-digits, so `--build-name=1.16.0-dev.764.ci7330+gfa725c28c` gets transformed into `--build-name=1.16.0.764.733072528`, which I don't think it's right, since the build name should be just 1.16.0 in this case, and the rest (or a part of it) should be the build number.
On Linux, I thought those flags are meaningless, but they do actually do something: they set the CMake variables `FLUTTER_VERSION_(MAJOR|MINOR|PATCH|BUILD)`, which are 4 independent numbers. To be honest, I don't know exactly the purpose of those, but its better to be safe then sorry and have those set correctly. And here, passing something like `--build-name=1.16.0 --build-number=dev.764.ci7330+gfa725c28c` simply prints a warning that the version could not be parsed and set those values to 1.0.0.0.
So the safe approach would be to pass correct arguments. For `--build-name`, the correct value is clear, it should be `1.16.0`. The question is then what number to pass to `--build-number`? I agree that my approach is not at all intuitive, and I did it just because that's what macos did. But what unique number could we derive for each build then?

Reply:
I am actually more concerned with the build name. Something like `1.16.0-dev.3557.pr698+g9bae52f58.win` is a *pre-release*, so it is actually lower than "1.16.0" (see https://semver.org/#spec-item-9 and the last example in https://semver.org/#spec-item-11). With this approach, that is lost entirely. I wonder if we should just set the version to 1.15.0 then...

These tools... not even support for semantic versioning :dotted_line_face:

Reply:
For posterity, after understanding better the meaning of the semantic version components, what we need is `--build-name=1.16.0-dev.3557.pr698` and `--build-number=decimal(9bae52f58)` which means that we take the commit hash and turn it from hexadecimal to decimal. This aligns most closely with semantic versioning, where stuff after `+` is considered build metadata. But since flutter only accepts a decimal number for that, converting the commit hash to decimal was a good compromise.